### PR TITLE
Improve doc example: Extracting the type parameter from a super-type

### PR DESCRIPTION
**Pull request by ArnoStrouwen**
Documentation describes the correct way of extracting the element type of a supertype:
https://docs.julialang.org/en/v1/manual/methods/#Extracting-the-type-parameter-from-a-super-type

However, one of the examples to showcase this is nonsensical since it is a union of multiple element types.
I have replaced this example with a union over the dimension.
Now, the `eltype_wrong` function still gives a similar error, yet the correct way returns the unambiguous answer.

### CODE REVIEW

File: doc/src/manual/methods.md [hunk 1]
```diff
@@ -698,11 +698,14 @@ While this works for declared types, it fails for types without
...
-Closest candidates are:
-  supertype(::DataType) at operators.jl:43
-  supertype(::UnionAll) at operators.jl:48
+julia> eltype_wrong(Union{Vector{Int}, Matrix{Int}})
+ERROR: MethodError: no method matching supertype(::Type{VecOrMat{Int64}})
+
+Closest candidates are:  
+  supertype(::UnionAll)  
+   @ Base operators.jl:44
+  supertype(::DataType)
```
Comment (LilithHafner) [inline doc/src/manual/methods.md:707]:
```suggestion
Closest candidates are: 
  supertype(::UnionAll)
   @ Base operators.jl:44
  supertype(::DataType)
```

**Review by LilithHafner — COMMENTED** [review]
Good catch, thanks!

File: doc/src/manual/methods.md (hunk 1 above)
Comment (nsajko) [reply doc/src/manual/methods.md:707]: There's still some trailing whitespace left after the suggested change :smile:

Comment (nsajko) [reply doc/src/manual/methods.md:707]: > `Closest candidates are: `

**Pull request merged**